tests/functional/s3/test_sync_command.py

```diff
@@ -164,3 +165,17 @@ def test_sync_with_delete_on_downloads(self):
         self.assertEqual(self.operations_called[0][0].name, 'ListObjects')
 
         self.assertFalse(os.path.exists(full_path))
+
+    def test_sync_skips_over_files_deleted_between_listing_and_transfer_attempt(self):
+        full_path = self.files.create_file('foo.txt', 'mycontent')
+        cmdline = '%s %s s3://bucket/' % (
+            self.prefix, self.files.rootdir)
+
+        # FileGenerator.list_files should skip over files that cause an OSError to be raised
+        # because they are missing when we try to get their stats.
+        with patch('awscli.customizations.s3.filegenerator.get_file_stat', side_effect=OSError()):
+            self.run_cmd(cmdline, expected_rc=0)
```

Comment:
Hm, should it cause the command to fail? I think if that is the case then my code does not resolve the issue I'm trying to resolve: https://github.com/aws/aws-cli/issues/1841

What we want is for these types of errors to be ignored. At least in my case, I have a script that runs after logrotate which syncs rotated logs to s3. Sometimes the logs get rotated away before they finish syncing – e.g. my pattern matches log1, log2, and log3, and while I am syncing log2, logrotate runs again and rotates log3 away, so by the time I get to it the file is gone and `s3 sync` fails. My script records when the `s3 sync` fails so that I have a record of possibly missing logs, so getting a return code of 2 in this case would not resolve my issue.